model/riscv_insts_vext_arith.sail

```diff
@@ -0,0 +1,2342 @@
+  then { handle_illegal(); return RETIRE_FAIL };
+
+  let 'n = num_elem;
+  let 'm = SEW;
+  let 'o = SEW_widen;
+
+  let vm_val  : vector('n, dec, bool)     = read_vmask(num_elem, vm, 0b00000);
+  let vd_val  : vector('n, dec, bits('m)) = read_vreg(num_elem, SEW, LMUL_pow, vd);
+  let rs1_val : bits('m)                  = get_scalar(rs1, SEW);
+  let vs2_val : vector('n, dec, bits('o)) = read_vreg(num_elem, SEW_widen, LMUL_pow_widen, vs2);
+  result      : vector('n, dec, bits('m)) = undefined;
+  mask        : vector('n, dec, bool)     = undefined;
+
+  (result, mask) = init_masked_result(num_elem, SEW, LMUL_pow, vd_val, vm_val);
+
+  assert(SEW_widen <= 64);
+  foreach (i from 0 to (num_elem - 1)) {
+    if mask[i] then {
+      let shift_amount = get_shift_amount(rs1_val, SEW_widen);
+      let rounding_incr = get_fixed_rounding_incr(vs2_val[i], shift_amount);
+      result[i] = match funct6 {
+        NX_VNCLIPU => {                    
+                        let result_wide = (vs2_val[i] >> shift_amount) + EXTZ('o, rounding_incr);
+                        unsigned_saturation('m, result_wide)
+                      },
+        NX_VNCLIP  => {       
+                        let v_double : bits('m * 4) = EXTS(vs2_val[i]);
+                        let result_wide = slice(v_double >> shift_amount, 0, 'o) + EXTZ('o, rounding_incr);
+                        signed_saturation('m, result_wide)
+                      }
+      }
+    }
+  };
+
+  write_vreg(num_elem, SEW, LMUL_pow, vd, result);
+  vstart = EXTZ(0b0);
+  RETIRE_SUCCESS
+}
+
+mapping nxtype_mnemonic : nxfunct6 <-> string = {
+  NX_VNCLIPU     <-> "vnclipu.wx",
+  NX_VNCLIP      <-> "vnclip.wx"
+}
+
+mapping clause assembly = NXTYPE(funct6, vm, vs2, rs1, vd)
+  <-> nxtype_mnemonic(funct6) ^ spc() ^ vreg_name(vd) ^ sep() ^ vreg_name(vs2) ^ sep() ^ reg_name(rs1) ^ maybe_vmask(vm)
+
+/* ***************** OPIVX (Vector Slide & Gather Instructions) ****************** */
+/* Slide and gather instructions extend rs1/imm to XLEN intead of SEW bits */
+union clause ast = VXSG : (vxsgfunct6, bits(1), regidx, regidx, regidx)
+
+mapping encdec_vxsgfunct6 : vxsgfunct6 <-> bits(6) = {
+  VX_VSLIDEUP     <-> 0b001110,
+  VX_VSLIDEDOWN   <-> 0b001111,
+  VX_VRGATHER     <-> 0b001100
+}
+
+mapping clause encdec = VXSG(funct6, vm, vs2, rs1, vd) if haveRVV()
+  <-> encdec_vxsgfunct6(funct6) @ vm @ vs2 @ rs1 @ 0b100 @ vd @ 0b1010111 if haveRVV()
+
+function clause execute(VXSG(funct6, vm, vs2, rs1, vd)) = {
+  let SEW_pow  = get_sew_pow();
+  let SEW      = get_sew();
+  let LMUL_pow = get_lmul_pow();
+  let VLEN_pow = get_vlen_pow();
+  let num_elem = get_num_elem(LMUL_pow, SEW);
+
+  if illegal_normal(vd, vm) then { handle_illegal(); return RETIRE_FAIL };
+
+  let 'n = num_elem;
+  let 'm = SEW;
+
+  let vm_val  : vector('n, dec, bool)     = read_vmask(num_elem, vm, 0b00000);
+  let rs1_val : nat                       = unsigned(X(rs1));
```

Comment:
This is because in this case we want to read rs1 as an integer value, and will only use it to compare element index. For other cases rs1 is read as an SEW width bit vector to be calculated with vs2 elements.